src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1410,7 +1671,13 @@ private void VisitObjectElementInitializer(int containingSlot, BoundAssignmentOp
         private new void VisitCollectionElementInitializer(BoundCollectionElementInitializer node)
         {
             // Note: we analyze even omitted calls
-            VisitArguments(node, node.Arguments, refKindsOpt: default, node.AddMethod, node.ArgsToParamsOpt, node.Expanded);
+            ImmutableArray<TypeSymbolWithAnnotations> results = VisitArguments(node, node.Arguments, refKindsOpt: default, node.AddMethod, node.ArgsToParamsOpt, node.Expanded);
+            _analyzedNullabilityMap[node] = results[0];
+            if (node.ImplicitReceiverOpt != null)
+            {
+                Debug.Assert(node.ImplicitReceiverOpt.Kind == BoundKind.ImplicitReceiver);
+                _analyzedNullabilityMap[node.ImplicitReceiverOpt] = TypeSymbolWithAnnotations.Create(node.ImplicitReceiverOpt.Type, NullableAnnotation.NotNullable);
```

Comment:
>_analyzedNullabilityMap[node.ImplicitReceiverOpt] = [](start = 16, length = 51)

If the receiver is implicit, who is using this map entry?

Reply:
When we get to the rewriting step, we'll attempt to set top-level nullability and it will use the entry. This will then be exposed concretely in the IOperation tree.

---
In reply to: [260059491](https://github.com/dotnet/roslyn/pull/33591#discussion_r260059491) [](ancestors = 260059491)